methcomp/regression.py

```diff
@@ -1,532 +1,279 @@
-                 point_kws):
-        self.method1: np.array = np.asarray(method1)
-        self.method2: np.array = np.asarray(method2)
-        self.x_title = x_label
-        self.y_title = y_label
-        self.graph_title = title
-        self.CI = CI
-        self.color_points = color_points
-        self.color_regr = color_regr
-        self.line_reference = line_reference
-        self.line_CI = line_CI
-        self.legend = legend
-        self.point_kws = {} if point_kws is None else point_kws.copy()
-
-        self._check_params()
-        self._derive_params()
-
-    def _check_params(self):
-        if len(self.method1) != len(self.method2):
-            raise ValueError('Length of method 1 and method 2 are not equal.')
-        if self.CI is not None and (self.CI > 1 or self.CI < 0):
-            raise ValueError('Confidence interval must be between 0 and 1.')
-        if any([not isinstance(x, str) for x in [self.x_title, self.y_title]]):
-            raise ValueError('Axes labels arguments should be provided as a str.')
-
-    def _derive_params(self):
-        self.n = len(self.method1)
-        _model = sm.OLS(self.method1, sm.add_constant(self.method2)).fit()
-        _params = _model.params
-        _confint = _model.conf_int(alpha=self.CI)
-
-        self.intercept = [_confint[0][0], _params[0], _confint[0][1]]
-        self.slope = [_confint[1][0], _params[1], _confint[1][1]]
-
-    def plot(self, ax):
-        # plot individual points
-        ax.scatter(self.method1, self.method2, s=20, alpha=0.6, color=self.color_points,
-                   **self.point_kws)
-
-        # plot reference line
-        if self.line_reference:
-            ax.plot([0, 1], [0, 1], label='Reference',
-                    color='grey', linestyle='--', transform=ax.transAxes)
-
-        # plot linear regression
-        _xvals = np.array(ax.get_xlim())
-        _yvals = [self.intercept[s] + self.slope[s] * _xvals for s in range(0, 3)]
-        ax.plot(_xvals, _yvals[0], label=f'{self.intercept[0]:.2f} + {self.slope[0]:.2f} * Method 1',
-                color=self.color_regr, linestyle='-')
-        ax.fill_between(_xvals, _yvals[1], _yvals[2], color=self.color_regr, alpha=0.2)
-        if self.line_CI:
-            ax.plot(_xvals, _yvals[1], linestyle='--')
-            ax.plot(_xvals, _yvals[2], linestyle='--')
-
-        if self.legend:
-            ax.legend(loc='upper left', frameon=False)
-
-        ax.set_ylabel(self.y_title)
-        ax.set_xlabel(self.x_title)
-        if self.graph_title is not None:
-            ax.set_title(self.graph_title)
-
-
-def linear(method1, method2,
-           x_label='Method 1', y_label='Method 2', title=None,
-           CI=0.95, line_reference=True, line_CI=False, legend=True,
-           color_points='#000000', color_regr='#008bff',
-           point_kws=None,
-           square=False, ax=None):
+    return PassingBablok(method1=method1, method2=method2, CI=CI).plot(
```

Comment:
see earlier comment about returning class vs class.plot()

Reply:
Awaiting answer for above :)